datafusion/core/src/execution/context.rs

```diff
@@ -958,6 +958,22 @@ impl SessionContext {
         }
     }
 
+    /// Return a [`TabelProvider`] for the specified table.
+    pub fn table_provider<'a>(
```

Comment:
👍  it might be nice to refactor `fn table()` to call this function now to avoid some duplication.